client/core/federer_handlers/cdr.py

```diff
@@ -72,6 +72,7 @@ def process_cdr(self, cdr_xml):
                 break
         if callee[0] == "+":
             callee = callee[1:]
+        hangupcause = get_tag_text(cdr_dom.getElementsByTagName("hangup_cause")[0].childNodes)
```

Comment:
Thanks for your contribution-- and for patiently waiting for a review
Please define this in a helper function that returns a default string "UNKNOWN" in <1% error case the field is missing and you get a list index error. We should still process the billing event regularly